sql/analyzer/indexes.go

```diff
@@ -347,7 +351,11 @@ func getComparisonIndexLookup(
 			var lookup sql.IndexLookup
 			switch e.(type) {
 			case *expression.Equals, *expression.NullSafeEquals:
-				lookup, err = sql.NewIndexBuilder(ctx, idx).Equals(ctx, normalizedExpressions[0].String(), value).Build(ctx)
```

Comment:
Does expression.IsNull belong here too?

Reply:
We convert IsNull to Equals(expr, nil) and recurse, I left it as is. I can undo that if you prefer.